### DISCUSSION TITLE
Install Homebrew on Mac M2

### DISCUSSION THREAD

**Discussion by AlexandreKang** [Homebrew/discussions#4341] [Getting started] [answered]
### Output of `brew config`

```shell
zsh: command not found: brew
```

### Output of `brew doctor`

```shell
zsh: command not found: brew
```

### Description of issue

I can't install Homebrew. I don't know why it asks for Git while others do not need to install git when installing Homebrew. I read the blogs and watch YouTube videos but can't find a solution.

Last login: Sun Mar 19 16:21:07 on console
alexandrekang@Alexandres-MacBook-Air ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R alexandrekang:admin /opt/homebrew
Unexpected Git version: ''!
You must install Git before installing Homebrew. See:
  https://docs.brew.sh/Installation
  
  Thank you
<img width="1470" alt="Screenshot 2023-03-19 at 4 57 56 PM" src="https://user-images.githubusercontent.com/33300743/226208541-1bda5ab4-82a0-4403-9945-a47a9cf6748e.png">

**Comment by p-linnane** [comment]
You need the Xcode Command Line Tools before you install `brew`. Run `xcode-select --install` first and try again.

**Reply by AlexandreKang** [reply]
I already install it
<img width="490" alt="Screenshot 2023-03-19 at 6 22 04 PM" src="https://user-images.githubusercontent.com/33300743/226213237-f4e7b7d6-4d91-4eac-ba38-c5c72c20b586.png">

<img width="870" alt="Screenshot 2023-03-19 at 6 23 31 PM" src="https://user-images.githubusercontent.com/33300743/226213448-0db6ceb5-2898-4a10-8941-0a1edc12355b.png">

**Reply by p-linnane** [reply]
What does `echo $PATH` output?

**Reply by AlexandreKang** [reply]
<img width="390" alt="Screenshot 2023-03-19 at 6 31 28 PM" src="https://user-images.githubusercontent.com/33300743/226213869-2c2ff79e-73a9-478e-b64e-f5a74170e8a0.png">

**Reply by gromgit** [reply]
Please post text output _as text_, not screenshots.

You're erasing your `PATH` somewhere. That's _never_ a good idea, as you're left with almost no ability to do anything at the command line, not even:
```sh
% cat test                                                                       
zsh: command not found: cat
```
To diagnose, do the following:
```sh
# Reset PATH to a sane minimum
eval "$(/usr/libexec/path_helper -s)"
# Trace zsh shell startup logic
script /tmp/shell.log zsh -xli -c "exit"
```
then post the output (as _text_, please!) of:
```sh
grep PATH= /tmp/shell.log
```

**Reply by AlexandreKang** [reply]
Hello, thank you

Here is the output of `grep PATH= /tmp/shell.log:`
```
+/etc/zprofile:7> eval 'PATH="/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin";' export 'PATH;'
+(eval):1> PATH=/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin 
```

**Reply by gromgit** [reply]
And `echo $PATH` still prints nothing?

**Reply by AlexandreKang** [reply]
It prints
```
alexandrekang@Alexandres-MacBook-Air ~ % echo $path
/usr/local/bin /System/Cryptexes/App/usr/bin /usr/bin /bin /usr/sbin /sbin
```

**Reply by gromgit** [reply]
Now try installing Homebrew again.

**Reply by AlexandreKang** [reply]
The output:
```
alexandrekang@Alexandres-MacBook-Air ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R alexandrekang:admin /opt/homebrew
Unexpected Git version: ''!
You must install Git before installing Homebrew. See:
  https://docs.brew.sh/Installation
alexandrekang@Alexandres-MacBook-Air ~ % 
```

**Reply by gromgit** [reply]
Sorry, thinking of the wrong shell. Do this:
```
PS4='+$BASH_SOURCE> ' script /tmp/shell.log /bin/bash -xli -c exit
```
then post the output of:
```
grep PATH= /tmp/shell.log
```

**Reply by AlexandreKang** [reply]
The output:
```
alexandrekang@Alexandres-MacBook-Air ~ % grep PATH= /tmp/shell.log
+/etc/profile> eval 'PATH="/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin";' export 'PATH;'
++/etc/profile> PATH=/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

**Comment by splaney** [comment, marked as answer]
Maybe someone with more experience can chime in here, but I found a bandaid solution that worked for me.

Opened up my first mac in a few years today and experienced the exact same thing with it not recognizing the git install from XCode, even though zsh was telling me git was installed.

Since the homebrew install script is a bash script, I opened a bash terminal and saw:
```
me@my-MBP ~ % bash
bash-3.2$ git
bash: /usr/local/bin/git: Bad CPU type in executable
```

A google search led me to [this stackexchange post](https://apple.stackexchange.com/questions/408375/zsh-bad-cpu-type-in-executable).

I then ran the following:

```
softwareupdate --install-rosetta
arch -x86_64 zsh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

However, now you have brew installed in x86, not ARM. I then found [this](https://github.com/Homebrew/discussions/discussions/1379), which after following those directions (new non-x86 terminal) I was able to install homebrew. Make sure to follow the instructions to add it to the path.

I also did the following afterwards:

```
me@my-MBP ~ % which git
/usr/local/bin/git
me@my-MBP ~ % rm -rf /usr/local/bin/git
me@my-MBP ~ % which git
/usr/bin/git
```

To remove an x64 git that was clobbering my attempts to use ARM apps. 

And, ta-da! Good to go! (for now, it seems) 

Sorry if I missed a few steps, was just as frustrated trying to get it all working!

**Reply by Anto1368** [reply]
I had the same problem.
Fixed deleting the git under `/usr/local/bin/`

**Reply by AlexandreKang** [reply]
Thanks a lot @Anto1368 ! It worked

**Reply by gromgit** [reply]
@AlexandreKang Just to confirm, you had a `/usr/local/bin/git` binary?

**Reply by AlexandreKang** [reply]
@gromgit yes

**Comment by StevenRoan** [comment]
Also come across this issue. 

TL:DR,  `ls -alh /usr/local/bin/git` and get the link of your previously brew-installed git folder and rm it. 

My path is `rm -rf /usr/local/Cellar/git  `

root cause----

The root cause is the homebrew install script will try to use /usr/local/bin/git, which is bad CPU in type in the migration case.

After rm it, brew will use the one in `/usr/bin/git` which is installed by x-code

**Comment by saikumar305** [comment]
Still facing the same issue on Mac M2 , didn't find any solution
